client/docs/service_groups.md

```diff
@@ -0,0 +1,160 @@
+<!---
+Licensed under Creative Commons Attribution 4.0 International License
+https://creativecommons.org/licenses/by/4.0/
+--->
+
+# Service Groups #
+
+Service groups provide a means to simplify and reuse specification of
+services, resources and policies for creating new contracts.
+
+Contract creation requires specification of enclave, provisioning, and
+stoage services; preferred or optimized services; and storage and
+replication policies. Rather than provide the information separately
+for every contract created, service groups define collections of
+services and policies for using those services that can be used to
+simplify contract creation.  For example, a service group for testing
+and developing new contracts can be created from services provided
+locally. Another group can be defined to for final production
+deployment.
+
+Service groups can be easily defined for all of the services
+associated with a given site (see the various `create_from_site`
+subcommands below). They can also be created for services provided by
+multiple sites (for greater resiliency and availability).
+
+All service groups have a name and a list of services (generally
+represented by the names or URLs for the services defined in the
+service endpoint database). Each type of service group (`eservice`,
+`pservice`, and `sservice`) has additional unique attributes that
+describe policies that accompany the group. For example, enclave
+services have a preferred service that identifies a place more likely
+to have contract state pre-cached.
+
+While the name space for each type of service group is independent
+(that is, you can have an enclave service group with the same name as
+a provisioning service group), for convenience we often use the same
+name for groups of different types when we use them together. For
+example, we may name enclave, provisioning and storage service groups
+`develop` because we always use them together for contract
+development.
+
+While groups can be specified for any operation that creates a
+contract, if no group is specified, then a group called 'default' will
+be used. That group must be created.
+
+**NOTE:** Services included in service groups must be stored in the
+service endpoint database (see `pdo-service-db` command). Generally
+this means that service endpoint must be added (usually by importing a
+site configuration file) before groups can be created.
+
+## Service Groups Data File ##
+
+Service group information is stored in a database file identified by
+the configuration variable `Service.GroupDatabaseFile` which defaults
+to `${PDO_HOME}/data/groups_db.mdb`. Command line switches allow for
+the use of alternative group files.
+
+## Commands for Working with Service Groups ##
+
+### Service Groups: `pdo-service-groups` ###
+
+The `pdo-service-groups` command provides operations over the collection
```

Comment:
afaik we translate these commands to psh-commands by stripping the `pdo-` but here we seem to also convert the `-` between service and groups to a `_` where in create_from_site we already use lower-cases.  Would be good to have a consistent convention, probably best call this here `pdo-service_groups` ?

Reply:
not really.

that being said... there is a bigger issue across all of PDO for being consistent about use of '-' or '_'. Not going to address this for the time being.